libcore/File.vala

```diff
@@ -0,0 +1,1177 @@
+public class GOF.File : GLib.Object {
```

Comment:
@jeremypw  Isn't this file expected to have a GPL header?

Reply:
Yes.